### PR TITLE
OSDOCS-11138: update RHDE page for Microshift 4.17

### PR DESCRIPTION
**Pull request by aravipra**
Version(s):
N/A (no CPs, version "4" is for Pantheon only; versionless branch in the repo)
Can be merged, but Pantheon sync is EMBARGOED for after OCP 4.17 GA when the rest of the docs are published.

Issue:
[OSDOCS-11138](https://issues.redhat.com/browse/OSDOCS-11138)

Link to docs preview:
[Red Hat Device Edge product release compatibility matrix](https://78616--ocpdocs-pr.netlify.app/openshift-rhde/latest/overview/rhde-overview.html)

Review:
- [x] QE has approved this change.
- [x] SME has approved this change.

See also https://github.com/openshift/openshift-docs/pull/78509 for an approved version of this table.

### CODE REVIEW

**Review by DanielFroehlich — APPROVED** [review]
/LGTM

File: modules/about-rhde.adoc [hunk 1]
```diff
@@ -43,6 +43,11 @@ The latest release notes for each product that is part of {product-title} are av
 ^|*{microshift} Release Status*
 ^|*Supported {microshift} Version&#8594;{microshift} Version Updates*
 
+^|9.4
+^|4.17
+^|Generally Available
+^|4.17.0&#8594;4.17.z and 4.16&#8594;4.17
```
Comment (dhensel-rh) [inline modules/about-rhde.adoc:49]: I am pretty sure there is a Y-2 scenario that should be included 

4.15 -> 4.17

Comment (DanielFroehlich) [reply modules/about-rhde.adoc:49]: Right! But officially and externally documented, we want to support this only for EUS -> EUS upgrades.

Comment (dhensel-rh) [reply modules/about-rhde.adoc:49]: Thanks for clarification @DanielFroehlich. Comment can be ignored.

**Review by skopacz1 — APPROVED** [review]
LGTM

**Comment by aravipra** [comment]
WIP do not merge

**Comment by mburke5678** [comment]
@aravipra As I read the description, I am merging this PR, but no cherry picking.

**Pull request merged**